### PR TITLE
Adding UserJourneyIdCache table

### PR DESCRIPTION
**Pull request by richa-misra**
Ticket Number: #TT2-1521 🎫

💡 Description
Provisioned dynamobDB table for enrichment layer - table created in new event-processing template deployed by new event-processing github workflow 

:frame_with_picture: Screenshots (if applicable)
Table created in dev:
Screenshot 2024-03-20 at 09 39 03

### CODE REVIEW

File: .github/workflows/package-rp-events.yaml [hunk 1]
```diff
@@ -20,10 +20,10 @@ jobs:
             ARTIFACT_BUCKET_NAME_SECRET: ARTIFACT_BUCKET_NAME_RP_EVENTS_DEV
             GH_ACTIONS_ROLE_ARN_SECRET: GH_ACTIONS_ROLE_ARN_RP_EVENTS_DEV
             SIGNING_PROFILE_SECRET: SIGNING_PROFILE_NAME_RP_EVENTS_DEV
-          - target: RP_EVENTS_BUILD
-            ARTIFACT_BUCKET_NAME_SECRET: ARTIFACT_BUCKET_NAME_RP_EVENTS_BUILD
-            GH_ACTIONS_ROLE_ARN_SECRET: GH_ACTIONS_ROLE_ARN_RP_EVENTS_BUILD
-            SIGNING_PROFILE_SECRET: SIGNING_PROFILE_NAME_RP_EVENTS_BUILD
+    #          - target: RP_EVENTS_BUILD
```
Comment (richa-misra) [inline .github/workflows/package-rp-events.yaml:23]: needs to be reverted before merge

File: infrastructure/rp-events/template.yaml [hunk 2]
```diff
@@ -140,6 +140,32 @@ Resources:
             Resource:
               - '{{resolve:ssm:DatabaseKmsKeyArn}}'
 
+  UserJourneyIdCacheTable:
```
Comment (zaydgds) [inline infrastructure/rp-events/template.yaml:143]: can we export table arn as

1. ssm param
2. cloudformation export

as we need to access it in event processing

**Review by zaydgds — CHANGES_REQUESTED** [review]

File: infrastructure/rp-events/template.yaml (hunk 2 above)
Comment (dannymadden-gds) [reply infrastructure/rp-events/template.yaml]: exported just as ssm param as discussed on call with Richa and Zayd

File: .github/workflows/package-event-processing.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+env:
+  AWS_REGION: eu-west-2
+
+jobs:
+  package:
+    strategy:
+      matrix:
+        include:
+          - target: EVENT_PROCESSING_DEV
+            ARTIFACT_BUCKET_NAME_SECRET: ARTIFACT_BUCKET_NAME_CORE_EVENT_PROCESSING_DEV
```
Comment (zaydgds) [inline .github/workflows/package-event-processing.yaml:20]: this bucket `ARTIFACT_BUCKET_NAME_CORE_EVENT_PROCESSING_DEV` is used to deploy the core infra `infrastructure/core/template.yaml`. 

we need to create 3 new secrets and deploy the terraform infrastructure that creates the dev platform deployment pipelines https://github.com/govuk-one-login/txma-pipelines

if we reuse the bucket, we will be overwritting the core infra

**Review by zaydgds — CHANGES_REQUESTED** [review]

**Review by eearcher-gds — APPROVED** [review]

**Pull request merged**